javascript/ql/src/semmle/javascript/frameworks/LazyCache.qll

```diff
@@ -39,10 +55,23 @@ module LazyCache {
 
     override PathExpr getImportedPath() { result = getArgument(0) }
 
+    private LazyCacheVariable getVariable() {
+      result = cache
+    }
+
+    pragma[noopt]
     override DataFlow::Node getImportedModuleNode() {
+      this instanceof LazyCacheImport and
       result = this.flow()
       or
-      result = cache.getAPropertyRead(getLocalAlias())
+      exists(LazyCacheVariable variable, Expr base, PropAccess access, string localName |
+        variable = getVariable() and
```

Comment:
Lets repeat the warning where the most tempting refactoring opportunity is located.
 
```suggestion
      exists(LazyCacheVariable variable, Expr base, PropAccess access, string localName |
        // To avoid recursion, this should not depend on `SourceNode`.
        variable = getVariable() and
```